ext/drreg/drreg.c

```diff
@@ -1924,18 +3059,22 @@ drreg_init(drreg_options_t *ops_in)
      * the max instead of summing.
      */
     if (ops_in->struct_size > offsetof(drreg_options_t, do_not_sum_slots)) {
-        if (ops_in->do_not_sum_slots) {
-            if (ops_in->num_spill_slots > ops.num_spill_slots)
-                ops.num_spill_slots = ops_in->num_spill_slots;
-        } else
-            ops.num_spill_slots += ops_in->num_spill_slots;
+        ops.num_spill_slots = get_updated_num_slots(
+            ops_in->do_not_sum_slots, ops.num_spill_slots, ops_in->num_spill_slots);
+        if (ops_in->struct_size > offsetof(drreg_options_t, num_spill_simd_slots)) {
+            ops.num_spill_simd_slots =
+                get_updated_num_slots(ops_in->do_not_sum_slots, ops.num_spill_simd_slots,
+                                      ops_in->num_spill_simd_slots);
+        }
         ops.do_not_sum_slots = ops_in->do_not_sum_slots;
     } else {
-        if (ops.do_not_sum_slots) {
-            if (ops_in->num_spill_slots > ops.num_spill_slots)
-                ops.num_spill_slots = ops_in->num_spill_slots;
-        } else
-            ops.num_spill_slots += ops_in->num_spill_slots;
+        ops.num_spill_slots = get_updated_num_slots(
+            ops.do_not_sum_slots, ops.num_spill_slots, ops_in->num_spill_slots);
+        if (ops_in->struct_size > offsetof(drreg_options_t, num_spill_simd_slots)) {
```

Comment:
Can't be <= offsetof(drreg_options_t, do_not_sum_slots) and > > offsetof(drreg_options_t, num_spill_simd_slots) right?

Reply:
No it can't. I add an assert.

Reply:
Done